geomesa-compute/src/main/scala/org/locationtech/geomesa/compute/spark/GeoMesaSpark.scala

```diff
@@ -187,6 +188,224 @@ object GeoMesaSpark extends LazyLogging {
     groupedByDay.map { case (date, iter) => (date, iter.size) }
   }
 
+  def shallowJoin(sc: SparkContext, coveringSet: RDD[SimpleFeature], data: RDD[SimpleFeature], key: String): RDD[SimpleFeature] = {
+    // Broadcast sfts to executors
+    val sfts = sc.broadcast(GeoMesaSparkKryoRegistrator.typeCache.values.map{ sft =>
+      (sft.getTypeName, SimpleFeatureTypes.encodeType(sft))
+    }.toArray)
+
+    data.foreachPartition{ iter =>
+      sfts.value.foreach{ case (name, spec) =>
+        val sft = SimpleFeatureTypes.createType(name, spec)
+        GeoMesaSparkKryoRegistrator.putType(sft)
+      }
+    }
+
+    // Broadcast covering set to executors
+    val broadcastedCover = sc.broadcast(coveringSet.collect)
+
+    // Key data by cover name
+    val keyedData = data.mapPartitions { iter =>
+      import org.locationtech.geomesa.utils.geotools.Conversions._
+
+      iter.flatMap { sf =>
+        // Iterate over covers until a match is found
+        val it = broadcastedCover.value.iterator
+        var container = ""
+        while (it.hasNext && container == "") {
+          val cover = it.next()
+          // If the cover's polygon contains the feature, set the container
+          if (cover.geometry.contains(sf.geometry)) {
```

Comment:
I could switch to intersects, but currently the loop stops when the first cover is found. I did that to be slightly more efficient, but I could change that too to handle other geometries.

Though in that case, is it okay to have features get "double counted"? I can see this, for example, if the LineStrings were roads, and a single road would be counted for each region it was in, but the total count would be off. I'm also not sure if there are examples of those geometries that would only need to be counted once across all covers.

I could just let the function be a parameter and default to this.